Add terraform tile integration for the Datadog Provider (#8606)
DataDog/integrations-core: terraform/README.md

# Agent Check: terraform

## Overview

> The Datadog Terraform provider allows you to interact with the Datadog API through a Terraform configuration. You can manage your Datadog resources, such as Dashboards, Monitors, Logs Configuration, etc., with this configuration.

Reviewer: ```suggestion
The Datadog Terraform provider allows you to interact with the Datadog API through a Terraform configuration. You can manage your Datadog resources, such as Dashboards, Monitors, Logs Configuration, etc, with this configuration.
```

Should we link to Terraform site in this overview?

## Setup

### Installation

The Datadog Terraform provider is available through the [Terraform Registry][1].

### Configuration

1. [Install Terraform][2]

> 2. Create a directory to contain the Terraform configuration files, for example: `terraform_config`

Reviewer: ```suggestion
2. Create a directory to contain the Terraform configuration files, for example: `terraform_config/`
```
makes the directory name more obvious imo

> 3. Create a `main.tf` file in `terraform_config` with the following content:

Reviewer: ```suggestion
3. Create a `main.tf` file in `terraform_config/` directory with the following content:
```

Author: (Edited your comment to include `the`)

    ```
    terraform {
      required_providers {
        datadog = {
          source = "DataDog/datadog"
        }
      }
    }

    # Configure the Datadog provider
    provider "datadog" {
      api_key = var.datadog_api_key
      app_key = var.datadog_app_key
    }
    ```

4. Run `terraform init`. This will initialize the directory for use with Terraform and pull the Datadog provider.

> 5. Create any `.tf` file in this directory and start creating Datadog resources. For example:

Reviewer: ```suggestion
5. Create any `.tf` file in the `terraform_config/` directory and start creating Datadog resources. For example:
```
